src/lkl/setup.c

```diff
@@ -1386,18 +1428,16 @@ void lkl_start_init()
 
     sgxlkl_mtu = sgxlkl_enclave->tap_mtu;
 
-    initialize_enclave_event_channel(
-        sgxlkl_enclave->shared_memory.enc_dev_config,
-        sgxlkl_enclave->shared_memory.evt_channel_num);
+    initialize_enclave_event_channel(shm->enc_dev_config, shm->evt_channel_num);
 
     // Register console device
-    lkl_virtio_console_add(sgxlkl_enclave->shared_memory.virtio_console_mem);
+    lkl_virtio_console_add(shm->virtio_console_mem);
 
     // Register network tap if given one
     int net_dev_id = -1;
-    if (sgxlkl_enclave->net_fd != 0)
-        net_dev_id = lkl_virtio_netdev_add(
-            sgxlkl_enclave->shared_memory.virtio_net_dev_mem);
+    if (shm->virtio_net_dev_mem)
+        // TODO: Add flag in enclave config to disable this?
```

Comment:
Yes, whether there is outside networking should be an attestable enclave config parameter.

Reply:
Some for the console? I'm afraid setting the shared memory for the console to `NULL` will break all tests...

Reply:
Ok, we can ignore this in this PR, but eventually we should add new parameters to the enclave config for both of these.